### PR TITLE
Change error message when CuPy is not installed

### LINKED ISSUE #1176: Change a method reporting an error when CuPy is not correctly installed
Currently CuPy raises `importError` with this error message:
```ImportError: cannot import name core```

This is because `core` module is a native module (i mean .so file), and the setup script cannot build it when CUDA environment is not correctly setup. This error doesn't represent its situation. We need to change its message.

### PR DESCRIPTION
**Pull request by unnonouno**
fix #1176

### CODE REVIEW

**Comment by unnonouno** [comment]
Which exception is appropriate?

**Comment by okuta** [comment]
I think that exception is appropriate.

I tested on my environment.
I got the following message.
```
% python -c "import cupy"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "cupy/__init__.py", line 16, in <module>
    raise RuntimeError(msg)
RuntimeError: CuPy is not correctly installed. Please check your environment, uninstall Chainer and reinstall it with `pip install chainer --no-cache-dir -vvvv`.
```

LGTM

File: cupy/__init__.py [hunk 1]
```diff
@@ -4,9 +4,19 @@
...
+try:
+    from cupy import core
+except ImportError:
+    # core is a c-extention module.
+    # When a user cannot import core, it represents that CuPy is not correctly
+    # built.
+    msg = ('CuPy is not correctly installed. Please check your environment, '
+           'uninstall Chainer and reinstall it with `pip install chainer '
+           '--no-cache-dir -vvvv`.')
+    raise RuntimeError(msg)
```
Comment (okuta) [inline cupy/__init__.py]: How about setting the original exception?

Comment (unnonouno) [inline cupy/__init__.py]: Oh, do you mean `RuntimeError(msg, e)`?

Comment (okuta) [inline cupy/__init__.py]: Yes.
I suggest using `six.reraise`.
https://pythonhosted.org/six/#six.reraise

We want the original stack-trace when we develop CuPy.

Comment (unnonouno) [inline cupy/__init__.py]: I see.

File: cupy/__init__.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
     msg = ('CuPy is not correctly installed. Please check your environment, '
            'uninstall Chainer and reinstall it with `pip install chainer '
            '--no-cache-dir -vvvv`.')
-    raise RuntimeError(msg)
+    raise six.reraise(RuntimeError, msg)
```
Comment (okuta) [inline cupy/__init__.py]: How about `raise six.reraise(RuntimeError, msg, sys.exc_info()[2])`?

File: cupy/__init__.py [hunk 3]
```diff
@@ -4,9 +4,19 @@
...
+try:
+    from cupy import core
+except ImportError:
+    # core is a c-extention module.
+    # When a user cannot import core, it represents that CuPy is not correctly
+    # built.
+    msg = ('CuPy is not correctly installed. Please check your environment, '
+           'uninstall Chainer and reinstall it with `pip install chainer '
+           '--no-cache-dir -vvvv`.')
+    raise six.reraise(RuntimeError, msg)
```
Comment (okuta) [inline cupy/__init__.py]: How about `raise six.reraise(RuntimeError, msg, sys.exc_info()[2])`?

Comment (unnonouno) [inline cupy/__init__.py]: It's a good idea.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**